repos/system_upgrade/el7toel8/libraries/repofileutils.py

```diff
@@ -35,12 +42,25 @@ def parse_repofile(repofile):
     return RepositoryFile(file=repofile, data=data)
 
 
+def get_repodirs():
+    """
+    Return all directories yum scans for repository files, if they exist.
+    By default, the possible paths on RHEL 7 should be:
+    ['/etc/yum.repos.d', '/etc/yum/repos.d', '/etc/distro.repos.d']
+
+    ATTENTION: Requires the dnf module to be present.
+    """
+    with dnf.base.Base() as base:
+        base.conf.read(priority=dnf.conf.PRIO_MAINCONFIG)
+        return [d for d in base.conf.reposdir if os.path.isdir(d)]
```

Comment:
That looks good. Just for the record, we do not need to take care of symlinks (e.g. case when `/etc/distro.repos.d` points to `/etc/yum.repos.d`; in such a case, we report many duplicates always, but same happens in case of dnf, so it is consistent behaviour and I believe it is not expected now to have symlinks there)

Reply:
Tested the behavior and indeed, dnf goes into symlinks too. I'll add -L but we'll also have to filter out duplicate directories.

Reply:
@drehak  This will not be needed as DNF is not doing that as well. That's the case described above. If you create symlink to one of other directories, dnf will report duplicates. From that point, I think it should be ok to do the same. But if you want to handle it, fix could be like this:
```suggestion
        return list({os.path.realpath(d) for d in base.conf.reposdir if os.path.isdir(d)})
```